Firestore/Source/API/FIRDocumentSnapshot.mm

```diff
@@ -262,11 +263,11 @@ - (id)convertedReference:(const FieldValue &)value {
   const DatabaseId &database = _snapshot.firestore()->database_id();
   if (refDatabase != database) {
     // TODO(b/32073923): Log this as a proper warning.
-    NSLog(@"WARNING: Document %@ contains a document reference within a different database "
-           "(%s/%s) which is not supported. It will be treated as a reference within the "
-           "current database (%s/%s) instead.",
-          self.reference.path, refDatabase.project_id().c_str(), refDatabase.database_id().c_str(),
-          database.project_id().c_str(), database.database_id().c_str());
+    LOG_WARN("Document %s contains a document reference within a different database (%s/%s) which "
+             "is not supported. It will be treated as a reference within the current database "
+             "(%s/%s) instead.",
+             _snapshot.CreateReference().Path(), refDatabase.project_id(),
+             refDatabase.database_id(), database.project_id(), database.database_id());
```

Comment:
Wait; you've added a new parameter. .. Oh, we were missing one before. I've gotten so used to gcc counting printf templates for me, that I had forgotten that's not automatic. :(

I think it could be though.

```c
#include <stdio.h>
#include <stdarg.h>

void simple_printf(const char* fmt, ...)
  __attribute__((format (printf, 1, 2)));

void simple_printf(const char* fmt, ...) {
  // implementation omitted
}

int main(void) {
  simple_printf("%d%c%f%f", 3, 'a', 1.999, 42.5, 'b');
}
```

```
$ g++ test.cc -Wall
test.cc:12:50: warning: data argument not used by
      format string [-Wformat-extra-args]
  ..."%d%c%f%f", 3, 'a', 1.999, 42.5, 'b');
     ~~~~~~~~~~                       ^
1 warning generated.
```

That makes use of gcc's format attribute (https://gcc.gnu.org/onlinedocs/gcc-3.2/gcc/Function-
Attributes.html), so is obviously compiler specific. I don't know if clang has something similar. But if even one compiler checks this for us, travis should catch these issues.

This is obviously unrelated to your change, but if we can slot this into LOG_WARN and friends somehow (and I haven't looked to see if that's feasible), it'd be pretty beneficial.

Reply:
I haven't added a format parameter nor an argument. The diff suggests that's the case but it's not. What changed is I removed `.c_str()` from the arguments.

Unfortunately, this isn't a printf format string so GCC's checking doesn't help. Each placeholder here is just `%s`.

Reply:
Oh, you're right; it wasn't the c_str(), I just missed a comma.

Even though it's not technically a printf format string, the count of arguments might still work? I might take a few minutes and experiment with it... see if I can get it to spit out something useful.